contracts/staking/test/protocol_fees.ts

```diff
@@ -0,0 +1,361 @@
+import { blockchainTests, constants, expect, LogDecoder } from '@0x/contracts-test-utils';
+import { StakingRevertErrors } from '@0x/order-utils';
+import { BigNumber } from '@0x/utils';
+import { LogWithDecodedArgs } from 'ethereum-types';
+
+import {
+    artifacts,
+    TestProtocolFeesContract,
+    TestProtocolFeesERC20ProxyContract,
+    TestProtocolFeesERC20ProxyTransferFromCalledEventArgs,
+} from '../src';
+
+// tslint:disable:no-unnecessary-type-assertion
+blockchainTests('Protocol Fee Unit Tests', env => {
+    // The accounts that will be used during testing.
+    let owner: string;
+    let exchange: string;
+    let nonExchange: string;
+    let makerAddress: string;
+    let payerAddress: string;
+
+    // The contract that will be used for testng `payProtocolFee` and `_unwrapETH`.
+    let protocolFees: TestProtocolFeesContract;
+    let proxy: TestProtocolFeesERC20ProxyContract;
+
+    // The log decoder that will be used to decode logs from TestProtocolFeesERC20Proxy.
+    let logDecoder: LogDecoder;
```

Comment:
The wrappers should automatically decode logs now, so we shouldn't ever need to use `LogDecoder` anymore (you can just pass `artifacts` in as the last parameter when deploying a contract). We should probably make a task to remove it from all tests.

Reply:
Ah, that's what I was missing. I didn't specify `artifacts`